engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -141,7 +142,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws CloudRuntimeException, AgentUnavailableException;
```

Comment:
@rafaelweingartner
When i said they will have to catch i meant, When a method declaration has one or more exceptions defined using throws clause then the method-call must handle all the defined exceptions. If he wants to handle it he will have to catch it, if he dose not want to handle he can bubble it up.